### PR TITLE
Fix crash with models from ReadyPlayerMe by adding extra validation.

### PR DESCRIPTION
**Pull request by daleglass**
This also adds test code for the GLTF loader. The test currently relies on files linked in #239 

It also introduces a small improvement to add warning and error counts to the model class, so that the test can check whether there were any issues with the model.

### CODE REVIEW

File: tests/model-serializers/src/ModelSerializersTests.cpp [hunk 1]
```diff
@@ -0,0 +1,119 @@
+//
+//  AABoxCubeTests.h
+//  tests/octree/src
+//
+//  Created by Brad Hefta-Gaub on 06/04/2014.
+//  Copyright 2014 High Fidelity, Inc.
```
Comment (JulianGro) [inline tests/model-serializers/src/ModelSerializersTests.cpp:6]: Looks like the header needs updating

Comment (daleglass) [reply tests/model-serializers/src/ModelSerializersTests.cpp]: Doh.

**Review by ksuprynowicz — APPROVED** [review]
All looks fine to me. The models used in the test are not included, but it would be impossible to include them due to copyright.

**Comment by daleglass** [comment]
@ksuprynowicz  Doesn't RPM generate avatars for free? They say it's a CC-NC-SA license.

Maybe we could upload some test avatars to a bucket, and have them downloaded for the test? I can make the test spit out some text on the licensing if that'd help.

**Comment by akamicah** [comment]
Tested on Linux. No crash

**Comment by JulianGro** [comment]
> @ksuprynowicz Doesn't RPM generate avatars for free? They say it's a CC-NC-SA license.
> 
> Maybe we could upload some test avatars to a bucket, and have them downloaded for the test? I can make the test spit out some text on the licensing if that'd help.

You could try adding the license using reuse:
```
reuse addheader --license=CC-BY-NC-4.0 --copyright="Wolfprint 3D OÜ" --year=2022 dude.glb
```
I assume the copyright info is also in the file, but reuse cannot read it.

**Pull request merged**